cmd/ocitool/appendlayer_cmd.go

```diff
@@ -129,23 +129,21 @@ func AppendLayersCmd(c *cli.Context) error {
 
 	log.WithField("base_desc", baseManifestDesc).Debugf("using as base")
 
-	layerPaths := c.StringSlice("layer")
+	layerPaths := c.Generic("layer").(*flagutil.KeyValueFlag).Map
 
 	layerProvider := &blob.Index{
 		Blobs: make(map[digest.Digest]string),
 	}
 
 	layerDescs := make([]ocispec.Descriptor, 0, len(layerPaths))
-	for _, lp := range layerPaths {
-		ld, reader, err := ociutil.CreateDescriptorFromFile(lp)
+	for layerPath, layerDescriptorPath := range layerPaths {
```

Comment:
OOPS.  Non-deterministic map iteration means the layers we carefully specify in order on the command line end up all topsy-turvy.  Fix incoming.